IMP-1 - Basing the TAF

Daniil281991 opened the pull request:
- Create properties file and properties reader
      - Create simple tests for logging in logging out
      - base API helper to provide setting up and tearing down

Daniil281991 committed:
IMP-1 - Base the TAF with POM structure
      - Create properties file and properties reader
      - Create simple tests for logging in logging out
      - base API helper to provide setting up and tearing down

Daniil281991 committed:
IMP-1 - Fix API request, add getting credentials from environment variables

Daniil281991 committed:
IMP-1 - Fix testUserCanDeleteRepository

Daniil281991 committed:
IMP-1 - update .gitignore

lizvarabei commented on src/main/java/com/pages/Header.java:
```diff
@@ -0,0 +1,27 @@
+public class Header {
+    Page page;
+
+    private final String userProfileButton = ".Header-item:nth-child(7) .Header-link";
```
please do not use numbers in selectors

Daniil281991 replied:
fixed

lizvarabei commented on src/main/java/com/pages/Header.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    public void logOut() {
+        if (page.isVisible(userProfileButton)) {
```
do not add this condition.

Daniil281991 replied:
fixed

lizvarabei commented on src/main/java/com/pages/HomePage.java:
```diff
@@ -0,0 +1,28 @@
+    Page page;
+
+    private final String mainPageHeaderLocator = "main>div.my-3>h1";
+    private final String createNewRepositoryButton = "article:nth-child(3) div:nth-child(1) > a";
```
the same for selectors. Too unstable selectors. Do not use number. Only if it is table

Daniil281991 replied:
fixed

lizvarabei commented on src/main/java/com/pages/RepositorySettingsPage.java:
```diff
@@ -0,0 +1,24 @@
+    Page page;
+
+    private final String deleteRepositoryButton = ".Box li:nth-child(4) summary";
+    private final String deletingConfirmationTextSelector = "li:nth-child(4) p:nth-child(2) > strong";
```
the same for selectors

Daniil281991 replied:
fixed

lizvarabei commented on src/test/java/com/config/config.properties:
```diff
@@ -0,0 +1,10 @@
+USER_NAME = ta-user-danchik
+REPOSITORY_NAME = new_repository
+
+SCREENSHOT_FOLDER_PATH = src/test/resources/screenshots
```
you don't need it in config. You will not change it and will not use it in CI/CD

Daniil281991 replied:
fixed

Daniil281991 committed:
IMP-1 - fix locators

Daniil281991 committed:
IMP-1 - Add entities for elements
      - Add Annotations for tests
      - Add inheritance for all pages from BasePage
      - Add using element entities in pages

Daniil281991 committed:
IMP-1 - Update test suite

Daniil281991 committed:
IMP-1 - Add Logging for elements using java.util.logging

Daniil281991 committed:
IMP-1 - Update logging with using org.apache.log4j.Logger
      - Add logging listener
